core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -350,11 +350,20 @@ private[spark] class AppStatusListener(
         val e = it.next()
         if (job.stageIds.contains(e.getKey()._1)) {
           val stage = e.getValue()
-          stage.status = v1.StageStatus.SKIPPED
-          job.skippedStages += stage.info.stageId
-          job.skippedTasks += stage.info.numTasks
-          it.remove()
-          update(stage, now)
+          if (v1.StageStatus.PENDING.equals(stage.status)) {
```

Comment:
Is there nothing to be done here if StageStatus is ACTIVE?

Reply:
In this PR, I am only trying to fix any issues which are caused by out-of-order events leaving the ones caused by dropped events.

So, with that information, onJobEnd event does not do anything for active stages. Active stages are updated in onTaskEnd event or onStageCompleted event.

Reply:
In the previous behaviour, it would have marked the stages that were ACTIVE as SKIPPED, which now will not happen here anymore.
It looks like the code in `onStageCompleted` may handle that in the `stage.status = event.stageInfo.failureReason match {` handles that. Is that the case?

Reply:
Yes, that updates the status of a stage. Additionally, completed stages are removed from liveStages list after that or onTaskEnd event.

Reply:
> In the previous behaviour, it would have marked the stages that were ACTIVE as SKIPPED

That sounds right, but that was also incorrect behavior. I think we need to take a look at all the "what happens when events are dropped" cases, and that's better done separately. It's much less likely for events to be dropped here nowadays...